pkg/dockerregistry/server/manifestservice.go

```diff
@@ -91,6 +91,17 @@ func (m *manifestService) Get(ctx context.Context, dgst digest.Digest, options .
 
 	manifest, err = m.repo.manifestFromImageWithCachedLayers(image, ref.Exact())
 
+	if _, err := m.manifests.Put(WithRepository(ctx, m.repo), manifest, options...); err != nil {
+		context.GetLogger(m.ctx).Errorf("unable to put manifest to storage: %v", err)
+	} else if len(image.DockerImageManifest) > 0 {
+		image.DockerImageManifest = ""
+		image.DockerImageConfig = ""
+
+		if _, err := m.repo.updateImage(image); err != nil {
```

Comment:
Could you call this as long as the *blob is present on the storage* **and** the *image contains the manifest*?

Right now it will be skipped based on just *blob is present on the storage*.